packages/react-composites/src/composites/common/SendBox.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React, { useMemo } from 'react';
+import { SendBoxStylesProps, SendBox as SimpleSendBox } from '@internal/react-components';
+import { usePropsFor } from '../ChatComposite/hooks/usePropsFor';
+import { ChatCompositeOptions } from '../ChatComposite';
+/* @conditional-compile-remove(rich-text-editor) */
+import { Suspense } from 'react';
+/* @conditional-compile-remove(file-sharing) */
+import { ChatAdapter } from '../ChatComposite';
+/* @conditional-compile-remove(file-sharing) */
+import { fileUploadsSelector } from '../ChatComposite/selectors/fileUploadsSelector';
+/* @conditional-compile-remove(file-sharing) */
+import { useSelector } from '../ChatComposite/hooks/useSelector';
+
+/* @conditional-compile-remove(rich-text-editor) */
+const RichTextSendBoxWrapper = React.lazy(() =>
+  import('./RichTextSendBoxWrapper').then((module) => ({ default: module.RichTextSendBoxWrapper }))
+);
+
+/**
+ * @private
+ */
+export type SendBoxProps = {
+  options?: ChatCompositeOptions;
+  styles?: SendBoxStylesProps;
+  /* @conditional-compile-remove(file-sharing) */
+  adapter: ChatAdapter;
+};
+
+/**
+ * @private
+ */
+export const SendBox = (props: SendBoxProps): JSX.Element => {
+  const {
+    options,
+    styles,
+    /* @conditional-compile-remove(file-sharing) */
+    adapter
+  } = props;
+
+  const sendBoxProps = usePropsFor(SimpleSendBox);
+
+  /* @conditional-compile-remove(file-sharing) */
+  const activeFileUploads = useSelector(fileUploadsSelector).files;
+
+  const sendBoxStyles = useMemo(() => {
+    return Object.assign({}, styles);
+  }, [styles]);
+
+  const simpleSendBox = useMemo(
+    () => (
+      <SimpleSendBox
+        {...sendBoxProps}
+        autoFocus={options?.autoFocus}
+        styles={sendBoxStyles}
+        /* @conditional-compile-remove(file-sharing) */
+        activeFileUploads={activeFileUploads}
+        /* @conditional-compile-remove(file-sharing) */
+        onCancelFileUpload={adapter.cancelFileUpload}
+      />
+    ),
+    [
+      sendBoxProps,
+      options,
+      sendBoxStyles,
+      /* @conditional-compile-remove(file-sharing) */ activeFileUploads,
+      /* @conditional-compile-remove(file-sharing) */ adapter
+    ]
+  );
+
+  // /* @conditional-compile-remove(rich-text-editor) */
+  if (options?.richTextEditor === true) {
```

Comment:
If Contoso passes in a RichTextEditorOptions instead of a boolean, would it pass this if statement with strict equal?

Reply:
Let me double check how this would be handled.

Reply:
Double checked and this is valid and how it's done for the API in CallCompositeOptions that follows this pattern.